grpc.go

```diff
@@ -81,9 +81,53 @@ func handleError(wait bool, err error) error {
 	return err
 }
 
-// Online resources, nbResources specifies the maximum number of resources to online.
-// If nbResources is <= 0 then there is no limit and all resources are connected.
+// Offline resources, nbResources specifies the maximum number of resources to online.
+// If nbResources is <= 0 then there is no limit and all resources are disconnected.
 // Returns the number of resources connected.
+func offlineResources(resource onlineResource, nbResources int32) (uint32, error) {
+	// Get the bitmap of the online cpus
+	onlineCpuBitmapPath := filepath.Join(resource.sysfsOnlinePath, "online")
+	status, err := ioutil.ReadFile(onlineCpuBitmapPath)
+	if err != nil {
+		return 0, err
+	}
+
+	// Get from the last online cpu
+	strArr := strings.Split(strings.Replace(string(status), ",", "-", -1), "-")
```

Comment:
These two lines are a bit "dense" - could you split them up for more clarity please? Also, are these guaranteed not to fail if a particular field/substring is not found?

Reply:
My original idea is to get the last online cpu. There might be several cpu list cases:
1. 0-8
2. 0,3,4-8
3. 0,2,4,6,8

step one is to replace all "," with "-". e.g. "0,3,4-8" -> "0-3-4-8"
step two is to get the last field which is splitted by "8", that is "8"

IIUC, this parsing process should not fail if the status contains the correct online cpu list.

Reply:
>  "0,3,4-8" -> "0-3-4-8"

`0,3` means cpu `0` and cpu `3`. It doesn't mean cpus `0`, `1`, `2` and `3` though right?

Reply:
yes,